src/containers/CreateSecret/CreateSecret.js

```diff
@@ -91,16 +91,24 @@ export /* istanbul ignore next */ class CreateSecret extends Component {
 
   handleChangeNamespace = e => {
     const stateVar = 'namespace';
-    const stateValue = e.selectedItem.text;
-    this.setState(prevState => {
-      const newInvalidFields = prevState.invalidFields;
-      if (validateInputs(stateValue, stateVar)) {
-        delete newInvalidFields[stateVar];
-      } else {
-        newInvalidFields[stateVar] = true;
-      }
-      return { [stateVar]: stateValue, invalidFields: newInvalidFields };
-    });
+    if (e.selectedItem !== null) {
+      const stateValue = e.selectedItem.text;
+      this.setState(prevState => {
+        const newInvalidFields = prevState.invalidFields;
+        if (validateInputs(stateValue, stateVar)) {
+          delete newInvalidFields[stateVar];
+        } else {
+          newInvalidFields[stateVar] = true;
+        }
+        return { [stateVar]: stateValue, invalidFields: newInvalidFields };
+      });
+    } else {
+      const stateValue = '';
+      this.setState(prevState => {
+        const newInvalidFields = prevState.invalidFields;
+        return { [stateVar]: stateValue, invalidFields: newInvalidFields };
+      });
+    }
```

Comment:
We could simplify this a lot if we mark the namespace as invalid when it's cleared. This would give earlier feedback to the user instead of letting them clear it only to show an error later when they try to submit the form.

```diff
diff --git a/src/containers/CreateSecret/CreateSecret.js b/src/containers/CreateSecret/CreateSecret.js
index 637716a..eaff6ca 100644
--- a/src/containers/CreateSecret/CreateSecret.js
+++ b/src/containers/CreateSecret/CreateSecret.js
@@ -91,7 +91,7 @@ export /* istanbul ignore next */ class CreateSecret extends Component {

   handleChangeNamespace = e => {
     const stateVar = 'namespace';
-    const stateValue = e.selectedItem.text;
+    const stateValue = e.selectedItem ? e.selectedItem.text : '';
     this.setState(prevState => {
       const newInvalidFields = prevState.invalidFields;
       if (validateInputs(stateValue, stateVar)) {
```

Reply:
If a user presses the `X` button to clear the namespace surely they would be doing this to select another namespace anyway

Reply:
Have done this 😸